app/src/main/java/com/example/peterchu/watplanner/coursedetail/CourseDetailPresenter.java

```diff
@@ -85,9 +92,21 @@ public void onFailure() {
                     });
             isFirstLoad = false;
         }
+        // retrieve from cache.
+        schedule = dataRepository.getCourseSchedules();
+    }
+
+    public void pause() {
+        // save to cache.
+        dataRepository.setCourseSchedules(schedule);
```

Comment:
cause when users add/remove a course in this page, the resulting schedule needs to be cached. Also, conflict detection happens in this page.